text-summarization/Dockerfile

```diff
@@ -17,6 +17,11 @@ RUN apt-get install -y git default-jre && pip3.6 install -r opennmt_requirements
 ADD requirements.txt /service_requirements.txt
 RUN pip3.6 install -r service_requirements.txt
 
+# Install snet daemon
+RUN curl -OL https://github.com/singnet/snet-daemon/releases/download/v0.1.0/snetd-0.1.0.tar.gz && \
+      tar xzvf snetd-0.1.0.tar.gz && \
+      cp snetd-linux-amd64 /usr/local/bin/snetd# Install snet daemon
```

Comment:
>cp snetd-linux-amd64 /usr/local/bin/snetd# Install snet daemon

cp snetd-linux-amd64 /usr/local/bin/snetd

Reply:
We can use the new daemon release: [snetd-0.1.1.tar.gz](https://github.com/singnet/snet-daemon/releases/download/v0.1.1/snetd-0.1.1.tar.gz)

I think this download must be done inside a folder (like `/tmp`, `/opt/singnet` or even `/opt/singnet`) and then it must be deleted.
In fact I think all these instructions must be in one of these folders.

What do you think about add the `git clone` of the repo inside Dockerfile too?

Reply:
Ok - I will make the first changes you suggest.

Using `git clone` for the repository that the Dockerfile is in is redundant. The repo contents already get uploaded to the docker daemon added as part of the docker context, so we should never `git clone` the original repo without a very good reason.

Reply:
My idea was to use `docker build` with your Dockerfile's GitHub path.
But I agree with your way too. So yes, the user must clone the repo just one time.

Some points, following your Dockerfile:
- Keeping your way, I think the `fetch_models.py` will be missing, right?
- What about the `snetd.config.json`?
- To run `buildproto.sh` we need `grpcio-tools==1.14.1` (it's not in `requirements.txt`)

One thing I just saw is that the `snetd` version `0.1.1` changed `--config-path` to just `--config`.

Reply:
Ah ok. Most build systems will check out the repo as the first step. I haven't seen many people building directly from a remote Dockerfile.

`fetch_models.py` - it is run outside of the docker build, either by the developer or by CircleCI, the downloaded models become part of the docker context. They can be big, so no need to download them for every docker build. Everything in the repo/docker context gets added as part of `ADD . /text-summarization`

The `snetd.config.json` file for the service is hosted on the server. It contains a private key so is not allowed to be in the git repo (we have already had our private keys exposed by people accidentally adding them... so best to never add files with sensitive information into the repo)

I will add `grpcio-tools` to the requirements. I copied the buildproto.sh from another repo that wasn't using virtualenv and so it was calling python3.6 and using my system's grpcio-tools

Thanks for the heads up about the `--config-path` argument changing.